src/AutoRest.CSharp/LowLevel/Output/OperationMethodChainBuilder.cs

```diff
@@ -590,7 +590,7 @@ private void AddParameter(string name, InputParameter inputParameter, CSharpType
                 return;
             }
 
-            var convenienceMethodParameter = BuildParameter(inputParameter);
+            var convenienceMethodParameter = BuildParameter(inputParameter, frameworkParameterType);
```

Comment:
why do we introduce this change?  I didn't find the place we use this change in the PR.

Reply:
It is to resolve issue https://github.com/Azure/autorest.csharp/issues/3754. In the cadl-ranch test, the operation is simple which only has one head parameter, and it is not need to generate convenience method. So it is not reflect the fix there.

Reply:
But I don't see the convenience generated. Shall we add one that could generate convenience in cadl-ranch?

Reply:
actually, the issue that convenience method does not recognize the framework type as protocol method does is a common defect, and it is not related to Conditional Reuqest. Conditional Request is just a case which has the problem.